CMakeLists.txt

```diff
@@ -714,7 +714,7 @@ else()
   endif()
 
   # linker
-  if (NOT SANITIZE AND NOT OSSFUZZ AND NOT (WIN32 AND (CMAKE_C_COMPILER_ID STREQUAL "GNU" AND CMAKE_C_COMPILER_VERSION VERSION_LESS 9.1)))
+  if (NOT SANITIZE AND NOT OSSFUZZ AND NOT (WIN32 AND (CMAKE_C_COMPILER_ID STREQUAL "GNU" AND (CMAKE_C_COMPILER_VERSION VERSION_LESS 9.1 OR NOT STATIC))))
```

Comment:
It would be helpful to update the comments here to match the motivation you've described in this PR (Windows dynamically-linked builds fail with PIE regardless of gcc version).

Reply:
Updated. Appended the comment.